javalin/src/main/java/io/javalin/core/routing/ParserState.kt

```diff
@@ -2,113 +2,79 @@ package io.javalin.core.routing
-    }
-
-    return parseAsPathSegment(segment, rawPath)
 }
 
 private fun parseAsPathSegment(segment: String, rawPath: String): PathSegment {
     var state: ParserState = ParserState.NORMAL
     val pathNameAccumulator = mutableListOf<Char>()
-    fun mapSingleChar(char: Char): PathSegment? {
-        when (state) {
-            ParserState.NORMAL -> {
-                return when (char) {
-                    '*' -> PathSegment.Wildcard
-                    '{' -> {
-                        state = ParserState.INSIDE_CURLY_BRACKETS
-                        null
-                    }
-                    '<' -> {
-                        state = ParserState.INSIDE_ANGLE_BRACKETS
-                        null
-                    }
-                    '}', '>' -> throw MissingBracketsException(
-                        segment,
-                        rawPath
-                    ) // cannot start with a closing delimiter
-                    else -> PathSegment.Normal(char.toString()) // the single characters will be merged later
-                }
+    fun mapSingleChar(char: Char): PathSegment? = when (state) {
+        ParserState.NORMAL -> when (char) {
+            '*' -> PathSegment.Wildcard
+            '{' -> {
+                state = ParserState.INSIDE_SLASH_IGNORING_BRACKETS
+                null
+            }
+            '<' -> {
+                state = ParserState.INSIDE_SLASH_ACCEPTING_BRACKETS
+                null
             }
-            ParserState.INSIDE_CURLY_BRACKETS -> {
-                return when (char) {
-                    '}' -> {
-                        state = ParserState.NORMAL
-                        val name = pathNameAccumulator.joinToString(separator = "")
-                        pathNameAccumulator.clear()
-                        PathSegment.Parameter.SlashIgnoringParameter(name)
-                    }
-                    '{', '<', '>' -> throw MissingBracketsException(
-                        segment,
-                        rawPath
-                    ) // cannot start another variable inside a variable
-                    // wildcard is also okay inside a variable name
-                    else -> {
-                        pathNameAccumulator += char
-                        null
-                    }
-                }
+            '}', '>' -> throw MissingBracketsException(
+                segment,
+                rawPath
+            ) // cannot start with a closing delimiter
+            else -> PathSegment.Normal(char.toString()) // the single characters will be merged later
+        }
+        ParserState.INSIDE_SLASH_IGNORING_BRACKETS -> when (char) {
+            '}' -> {
+                state = ParserState.NORMAL
+                val name = pathNameAccumulator.joinToString(separator = "")
+                pathNameAccumulator.clear()
+                PathSegment.Parameter.SlashIgnoringParameter(name)
```

Comment:
maybe use the `createSlashIgnoringParam` function here?

Reply:
Yes!